### PR TITLE
feat: PreflightWorkflow

### PR DESCRIPTION
**Pull request by drymarau**
This PR introduces a new start destination in navigation graph - preflight.

It somewhat follows the existing UX of Pexip Infinity Connect apps and will enable call transfer handling in the near future.

### CODE REVIEW

File: sample/src/main/kotlin/com/pexip/sdk/sample/preflight/PreflightScreen.kt [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                }
+            }
+        }
+    }
+    if (rendering.childRendering != null) {
+        WorkflowRendering(
+            rendering = rendering.childRendering,
+            viewEnvironment = environment,
+        )
+    }
```
Comment (tpettersen) [inline sample/src/main/kotlin/com/pexip/sdk/sample/preflight/PreflightScreen.kt:72]: It is somewhat annoying that I lose state pressing outside the popups by accident. E.g: I introduce myself, write the meeting uri + host and the touches the screen outside of the popup when pin challenge is presented. Then I have to start all over again. That is a bit unfortunate

File: sample/src/main/kotlin/com/pexip/sdk/sample/SampleWorkflow.kt [hunk 2]
```diff
@@ -3,79 +3,130 @@ package com.pexip.sdk.sample
...
+
+    private fun RenderContext.createMicrophoneAudioTrackSideEffect(screen: SampleDestination) {
+        if (screen != SampleDestination.Permissions) runningSideEffect("createMicrophoneAudioTrackSideEffect") {
+            val track = localAudioTrackFactory.createLocalAudioTrack()
+            val action = OnMicrophoneAudioTrackChange(track)
+            actionSink.send(action)
+        }
+    }
+
+    private fun RenderContext.cameraVideoTrackSideEffect(track: CameraVideoTrack?) {
```
Comment (tpettersen) [inline sample/src/main/kotlin/com/pexip/sdk/sample/SampleWorkflow.kt:106]: There is a lifecycle issue with the camera:
- Open preflight
- Put the app in the background
- Open the camera app on your phone
- Open Video SDK app again

The camera is now gone, and you will not get it back when entering a conference even. The app needs to restart to get back to normal.

**Review by tpettersen — CHANGES_REQUESTED** [review]
The UI looks sharp 👍 Found a couple of issues

File: sample/src/main/kotlin/com/pexip/sdk/sample/SampleWorkflow.kt [hunk 3]
```diff
@@ -80,8 +84,8 @@ class SampleWorkflow @Inject constructor(
         }
     }
 
-    private fun RenderContext.createCameraVideoTrackSideEffect(screen: SampleDestination) {
-        if (screen != SampleDestination.Permissions) runningSideEffect("createCameraVideoTrackSideEffect") {
+    private fun RenderContext.createCameraVideoTrackSideEffect(count: UInt) {
+        if (count > 0u) runningSideEffect("createCameraVideoTrackSideEffect($count)") {
```
Comment (tpettersen) [inline sample/src/main/kotlin/com/pexip/sdk/sample/SampleWorkflow.kt:88]: This _works_, but I must admit that I am not to fan of counters here. 😅 

Just to have the discussion, what if you added `onCameraConnected(deviceName: String)` in `CameraVideoTrack.Callback`. Then you could store that deviceName in the state. 

`OnCameraDisconnected` sets the track to null, but you would still have the `deviceName`. So using `createCameraVideoTrack(deviceName, callback)` when the app is in the background would throw `IllegalStateException` if it is not available, and would not produce more renderings. The next time the app opens again, this should now work

Comment (drymarau) [reply sample/src/main/kotlin/com/pexip/sdk/sample/SampleWorkflow.kt:88]: I don't think it's a good approach as the `deviceName` may not necessarily be static and overall feels way more complex to track application lifecycle.

Comment (drymarau) [reply sample/src/main/kotlin/com/pexip/sdk/sample/SampleWorkflow.kt:88]: > Huh, could the device name of the current camera actually change runtime? 🤔

If it's an external camera it's definitely possible, there are no guarantees about `deviceName` not changing.

Use cases differ and I don't think it's feasible to provide Android lifecycle integration at all - it's fairly easy, if required, to build one yourself.

Comment (drymarau) [reply sample/src/main/kotlin/com/pexip/sdk/sample/SampleWorkflow.kt:88]: As for in-call, general advice is to not open camera app while in a conference call.

Comment (tpettersen) [reply sample/src/main/kotlin/com/pexip/sdk/sample/SampleWorkflow.kt:88]: ok 👌

**Pull request merged**